### PR TITLE
Add the possibilty to give an ssh key instead of password when connec…

### PR DESCRIPTION
**Pull request by AliceAlbano**
…ting to the source machine for a scan

### CODE REVIEW

File: hammr/commands/scan/scan.py [hunk 1]
```diff
@@ -493,10 +469,13 @@ def help_delete(self):
...
+        password = None
 
-        if not args.password:
-            password = getpass.getpass('Password for %s@%s: ' % (username, hostname))
-        else:
+        if not id_file:
+            if not args.password:
+                password = getpass.getpass('Password for %s@%s: ' % (username, hostname))
+        if args.password:
             password = args.password
```
Comment (CanutMIAGE) [inline hammr/commands/scan/scan.py]:
```
if not args.password:
                password = getpass.getpass('Password for %s@%s: ' % (username, hostname))
else:
                password = args.password
```
Is it not better that way? One checked removed

File: hammr/commands/scan/scan.py [hunk 2]
```diff
@@ -536,3 +516,44 @@ def get_uforge_auth(self, uforge_apikeys, uforge_password):
...
+    def print_scan_run_result_status(self, scan_status):
+        if scan_status.error:
+            printer.out("Scan  error: " + scan_status.message + "\n" + scan_status.errorMessage,
+                        printer.ERROR)
+            if scan_status.detailedError:
+                printer.out(scan_status.detailedErrorMsg)
+        elif scan_status.cancelled:
+            printer.out("Scan cancelled: " + scan_status.message, printer.WARNING)
+        else:
+            printer.out("Scan successfully", printer.OK)
```
Comment (CanutMIAGE) [inline hammr/commands/scan/scan.py]: "Scan successfully"
Scan successful ?

File: hammr/commands/scan/scan.py (hunk 1 above)
Comment (AliceAlbano) [reply hammr/commands/scan/scan.py]: but in this case you do not handle the case where you give an identity file (ssh key) and you don't need to ask for a password

File: hammr/commands/scan/scan.py (hunk 2 above)
Comment (AliceAlbano) [reply hammr/commands/scan/scan.py]: Haha yes, this is old code I guess

File: hammr/commands/scan/scan.py (hunk 1 above)
Comment (CanutMIAGE) [reply hammr/commands/scan/scan.py]: Yes you are right ;)

**Pull request merged**